### PR TITLE
Option for disable role joining

### LINKED ISSUE #249: Role joining is not model agnostic
We tried replacing flask-security in our project, however our project's User model does not have a sqlalchemy relationship for roles, it's a hybrid_property method on the class.

The changes from https://github.com/jwag956/flask-security/commit/32968e593d2f7c82ed2767b1e0d7927f3edfffaa assumes that if `roles` is defined as an attribute on the User class that it must be a relationship and thus joinable, which isn't always a safe assumption. 

I'm not sure what the best option here is outside of testing to see if the `roles` attribute is specifically a `sqlalchemy.orm.relationship` prior to forcing the relationship to be joined or if in our case we need to not have a `roles` attribute and override something somewhere on how the datastore gets roles for a user.

### PR DESCRIPTION
**Pull request by pffs**
Add configuration option to disable changing the loading method for the role attribute on the UserModel.

Closes #249

### CODE REVIEW

File: docs/configuration.rst [hunk 1]
```diff
@@ -318,6 +318,14 @@ Core - rarely need changing
 
     Default: ``roles``.
 
+.. py:data:: SECURITY_JOIN_USER_ROLES
+
+    Specifies whether to set the ``UserModel.roles`` loading relationship to ``joined`` when a ``roles`` attribute
+    is present for a SQLAlchemy Datastore. This restores pre 3.3.0 behavior and is required if the ``roles`` attribute
```
Comment (jwag956) [inline docs/configuration.rst]: 'Setting this to False... restores pre-3.3.0 behavior'

Also- please add a .. versionadded:: 3.4.0

File: flask_security/datastore.py [hunk 2]
```diff
@@ -324,13 +324,16 @@ def __init__(self, db, user_model, role_model):
...
     def get_user(self, identifier):
+        from flask import current_app
         from sqlalchemy import func as alchemyFn
         from sqlalchemy import inspect
         from sqlalchemy.sql import sqltypes
         from sqlalchemy.dialects.postgresql import UUID as PSQL_UUID
 
         user_model_query = self.user_model.query
-        if hasattr(self.user_model, "roles"):
+        if current_app.config["SECURITY_JOIN_USER_ROLES"] and hasattr(
```
Comment (jwag956) [inline flask_security/datastore.py]: to be consistent with rest of code - please use utils.config_value("JOIN_USER_ROLES")...

File: docs/configuration.rst (hunk 1 above)
Comment (pffs) [reply docs/configuration.rst]: Added!

File: flask_security/datastore.py (hunk 2 above)
Comment (pffs) [reply flask_security/datastore.py]: Changed.

**Pull request merged**